hyper/http11/connection.py

```diff
@@ -93,11 +105,16 @@ def connect(self):
         :returns: Nothing.
         """
         if self._sock is None:
-            sock = socket.create_connection((self.host, self.port), 5)
+            if not self.proxy_host:
+                host = self.host
+            else:
+                port = self.proxy_host
+                
+            sock = socket.create_connection((host, port), 5)
             proto = None
 
             if self.secure:
-                sock, proto = wrap_socket(sock, self.host, self.ssl_context)
+                sock, proto = wrap_socket(sock, host, self.ssl_context)
```

Comment:
This is not necessarily quite right.

The 'host' part of `wrap_socket` is used to validate TLS connections, which means it should always be the remote host, not the proxy host. Regardless, I don't think we'll support TLS connections directly to proxies without the CONNECT verb just yet. For that reason you might just want to temporarily turn this function off (`assert not self.proxy_host` in this block).